### ISSUE TITLE
Implement From<u32> etc for MetadataValue<Ascii>

### ISSUE THREAD

**Issue by adamchalmers** [hyperium/tonic#986]
### Crates

- tonic

### Motivation

Integers can be infallibly encoded into ASCII, and so they can be infallibly used as metadata values. But right now there's no API for this -- you have to turn the integer into a string, then use the fallible `MetadataValue::from_str` method.

Additonally, there's many places where the docs say "This function is intended to be replaced in the future by a TryFrom implementation once the trait is stabilized in std" e.g. [MetadataValue::from_str](https://docs.rs/tonic/latest/tonic/metadata/struct.MetadataValue.html#method.from_str). But TryFrom is stable now, so let's implement that.

### Proposal
MetadataValue should:
 - Impl From<all float and integer types, bools> 
 - Impl TryFrom<&str, String>

Then we should add a deprecation to the special "try_from" or "from" methods that currently exist, which these new From/TryFrom impls replace.

### Alternatives

We could leave the current methods in place, but that's a bit less convenient for programmers.

**Comment by LucioFranco** [comment]
Yeah, I would totally accept a PR for this. I think we should go the route of adding in the new APIs and deprecate the old ones. I think if http also has the same thing we can also do a PR for that. Feel free to tag me in the PR and ill take a look. Thanks!

**Comment by adamchalmers** [comment]
Actually, I'm confused, from reading the code it seems y'all are [using the same from_integers!](https://github.com/hyperium/tonic/blob/4b0ece6d2854af088fbc1bdb55c2cdd19ec9bb92/tonic/src/metadata/value.rs#L471) calls that the http crate does. This expands into generating a bunch of From impls for all the numeric types. Not sure why they don't show up in the docs.
